botocore/awsrequest.py

```diff
@@ -289,6 +290,7 @@ def prepare_request_dict(request_dict, endpoint_url, user_agent=None):
         else:
             url += '&%s' % encoded_query_string
     r['url'] = url
+    r['context'] = context or {}
```

Comment:
Whenever you have a default argument of None, you should always use `if context is not None`.  Especially for dicts and lists which have meaningful false values.